lib/ansible/modules/cloud/vmware/vmware_object_role_permission.py

```diff
@@ -0,0 +1,227 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# Copyright: (c) 2018, Derek Rushing <derek.rushing@geekops.com>
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+
+__metaclass__ = type
+
+ANSIBLE_METADATA = {
+    'metadata_version': '0.1',
+    'status': ['preview'],
+    'supported_by': 'community'
+}
+
+DOCUMENTATION = '''
+---
+module: vmware_object_role_permission
+short_description: Manage object permissions on an ESXi host
+description:
+    - Manage local roles on an ESXi host
+version_added: "2.5"
+author:
+- Derek Rushing (@kryptsi) <drush@kryptsi.com>
+notes:
+    - Tested on ESXi 6.5
+    - Be sure that the ESXi user used for login, has the appropriate rights to administer permissions
+requirements:
+    - "python >= 2.7"
+    - PyVmomi
+options:
+    role:
+        description:
+            - The role to be assigned permission.
+        required: True
+    principal:
+        description:
+            - The user to be assigned permission.
```

Comment:
Add a note that principal parameter is required if group is not present and same for group parameter.